### PR TITLE
chore: apply naming conventions to database resources in helm chart

### LINKED ISSUE #104: Adjust Database templates in Helm Chart to the defined naming conventions
## Helm Chart Naming Conventions

* The resource name must not contain the `kind` of the resource in the name. For example, the database secret shouldn't be `kubearchive-database-secret` and should be `kubearchive-database` instead.
* The resource names must be fully templated with no hardcoded values.

To prevent having several variables in a name resource like: {{ Release.Name }}-{{ .Values.sink.name }} we will take advantage of the [tpl function](https://helm.sh/docs/howto/charts_tips_and_tricks/#using-the-tpl-function):

```yaml
# values
operator: 
  name: "{{ .Release.Name }}-operator"

# template
{{ tpl .Values.operator.name . }}

# output
kubearchive-operator
```

## Goal

Review and update the Helm Chart templates related to the kubearchive database component to comply with the naming conventions.

### PR DESCRIPTION
**Pull request by mafh314**
Closes #104

### CODE REVIEW

**Review by beerparty — APPROVED** [review]

File: charts/kubearchive/templates/database/ps-claim.yaml [hunk 1]
```diff
@@ -4,13 +4,13 @@ apiVersion: v1
 kind: PersistentVolumeClaim
 metadata:
   labels:
-    app: {{ .Values.database.name }}
-  name: {{ .Values.database.name | default "postgresql" }}-volume-claim
-  namespace: {{ .Values.kubearchive.namespace }}
+    app: {{ tpl .Values.database.name . }}
+  name: {{ tpl .Values.database.name . }}
+  namespace: {{ .Release.Namespace }}
```
Comment (rh-hemartin) [inline charts/kubearchive/templates/database/ps-claim.yaml:9]: I think we decided to remove the namespaces. Also, this line is redundant with the default behaviour. If no namespace is defined, the namespace is the defined by the release.

Comment (maruiz93) [reply charts/kubearchive/templates/database/ps-claim.yaml:9]: +1 to the line being redundant. One of the points defined in the naming conventions is that we will remove the namespace of the resource when they are "kubearchive" as we will be providing it from the `-n` option on the `helm install` command.

But I don't think we agreed on removing the spaces. I think you might be referring to the issue https://github.com/kubearchive/kubearchive/issues/67, but as Sam mentions, `{{ variable }}` is the preferred Helm syntax. Also, note that this won't be a problem from the YAML linter POV as the `yamllint` will be applied after the `helm template` command render the values on the `charts/generated/` directory.

File: charts/kubearchive/templates/database/ps-deployment.yaml [hunk 2]
```diff
@@ -4,32 +4,32 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   labels:
-    app: {{ .Values.database.name | default "postgresql" }}
-  name: {{ .Values.database.name | default "postgresql" }}-deployment
-  namespace: {{ .Values.kubearchive.namespace }}
+    app: {{ tpl .Values.database.name . }}
+  name: {{ tpl .Values.database.name . }}
+  namespace: {{ .Release.Namespace }}
```
Comment (maruiz93) [inline charts/kubearchive/templates/database/ps-deployment.yaml:9]: I suggest removing this based on the fact that the `kubearchive` namespace is provided by the `-n` option in `helm install` command.
```suggestion
```

File: charts/kubearchive/templates/database/ps-pv.yaml [hunk 3]
```diff
@@ -5,13 +5,13 @@ kind: PersistentVolume
 metadata:
   labels:
     type: local
-    app: {{ .Values.database.name | default "postgresql" }}
-  name: {{ .Values.database.name | default "postgresql" }}-volume
-  namespace: {{ .Values.kubearchive.namespace }}
+    app: {{ tpl .Values.database.name . }}
+  name: {{ tpl .Values.database.name . }}
+  namespace: {{ .Release.Namespace }}
```
Comment (maruiz93) [inline charts/kubearchive/templates/database/ps-pv.yaml:10]: I suggest removing this based on the fact that the `kubearchive` namespace is provided by the `-n` option in `helm install` command.

```suggestion
```

File: charts/kubearchive/templates/database/ps-secret.yaml [hunk 4]
```diff
@@ -5,9 +5,9 @@ kind: Secret
 type: Opaque
 metadata:
   labels:
-    app: {{ .Values.database.name | default "postgresql" }}
-  name: {{ .Values.database.name | default "postgresql"}}-secret
-  namespace: {{ .Values.kubearchive.namespace }}
+    app: {{ tpl .Values.database.name . }}
+  name: {{ tpl .Values.database.name . }}
+  namespace: {{ .Release.Namespace }}
```
Comment (maruiz93) [inline charts/kubearchive/templates/database/ps-secret.yaml:10]: I suggest removing this based on the fact that the `kubearchive` namespace is provided by the `-n` option in `helm install` command.

```suggestion
```

File: charts/kubearchive/templates/database/ps-service.yaml [hunk 5]
```diff
@@ -4,12 +4,12 @@ apiVersion: v1
 kind: Service
 metadata:
   labels:
-    app: {{ .Values.database.name | default "postgresql" }}
-  name: {{ .Values.database.name | default "postgresql" }}
-  namespace: {{ .Values.kubearchive.namespace }}
+    app: {{ tpl .Values.database.name .  }}
+  name: {{ tpl .Values.database.name . }}
+  namespace: {{ .Release.Namespace }}
```
Comment (maruiz93) [inline charts/kubearchive/templates/database/ps-service.yaml:9]: I suggest removing this based on the fact that the `kubearchive` namespace is provided by the `-n` option in `helm install` command.

```suggestion
```

File: database/README.md [hunk 6]
```diff
@@ -29,7 +29,7 @@ Install [DBeaver Community](https://dbeaver.io/).
 These queries can be executed from the database.
 Connect to the database.
 ```bash
-$ kubectl exec -it [postgres_pod] -n [namespace] -- psql -h localhost -U [db_user] --password -p 5432 [db_name]
+$ kubectl exec -it [postgres_pod] -n kubearchive -- psql -h localhost -U [db_user] --password -p 5432 [db_name]
```
Comment (maruiz93) [inline database/README.md:32]: To make it easier for the reader, we can provide a way to automatically get the db pod based on the current naming:
```suggestion
$ kubectl exec -it  $(kubectl get -n kubearchive pods --no-headers -o custom-columns=":metadata.name" | grep database) -n kubearchive -- psql -h localhost -U ps_user --password -p 5432 postgresdb
```

**Review by maruiz93 — CHANGES_REQUESTED** [review]

File: charts/kubearchive/templates/database/ps-claim.yaml (hunk 1 above)
Comment (mafh314) [reply charts/kubearchive/templates/database/ps-claim.yaml]: removed the namespace`

File: charts/kubearchive/templates/database/ps-deployment.yaml (hunk 2 above)
Comment (mafh314) [reply charts/kubearchive/templates/database/ps-deployment.yaml]: removed the `namespace`

File: charts/kubearchive/templates/database/ps-pv.yaml (hunk 3 above)
Comment (mafh314) [reply charts/kubearchive/templates/database/ps-pv.yaml]: removed the `namespace`

File: charts/kubearchive/templates/database/ps-secret.yaml (hunk 4 above)
Comment (mafh314) [reply charts/kubearchive/templates/database/ps-secret.yaml]: removed the `namespace`

File: charts/kubearchive/templates/database/ps-service.yaml (hunk 5 above)
Comment (mafh314) [reply charts/kubearchive/templates/database/ps-service.yaml]: removed the `namespace`

File: database/README.md (hunk 6 above)
Comment (mafh314) [reply database/README.md]: Changed

**Review by rh-hemartin — APPROVED** [review]

**Review by maruiz93 — APPROVED** [review]

**Comment by rh-hemartin** [comment]
I think that is failing the mix between using `Release.Name` as name and also using a define function in `_helpers.tpl`. I suggest you remove the define function and follow the same approach as the secret name for the api.

**Review by rh-hemartin — APPROVED** [review]

**Pull request merged**